gnovm/pkg/gnolang/op_eval_test.go

```diff
@@ -0,0 +1,75 @@
+package gnolang_test
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+)
+
+// parseIntegerLiteral function is a Mock-up of the doOpEval() function in `op_eval.go`.
+func parseIntegerLiteral(value string) (*big.Int, error) {
+	if len(value) > 2 && value[0] == '0' {
+		bi := big.NewInt(0)
+		var ok bool
+		switch value[1] {
+		case 'b', 'B':
+			_, ok = bi.SetString(value[2:], 2)
+		case 'o', 'O':
+			_, ok = bi.SetString(value[2:], 8)
+		case 'x', 'X':
+			_, ok = bi.SetString(value[2:], 16)
+		case '0', '1', '2', '3', '4', '5', '6', '7':
+			_, ok = bi.SetString(value, 8)
+		default:
+			ok = false
+		}
+		if !ok {
+			return nil, fmt.Errorf("invalid integer constant: %s", value)
+		}
+		return bi, nil
+	} else {
+		bi, ok := new(big.Int).SetString(value, 10)
+		if !ok {
+			return nil, fmt.Errorf("invalid integer constant: %s", value)
+		}
+		return bi, nil
+	}
+}
+
+type testCase struct {
+	input    string
+	expected string
+	hasError bool
+}
+
+func TestParseIntegerLiteral(t *testing.T) {
+	testCases := []testCase{
+		{"012345", "5349", false},
+		{"02001", "1025", false},
+		{"002001", "1025", false},
+		{"0002001", "1025", false},
+		{"0o12345", "5349", false},
+		{"0x1a2b", "6699", false},
+		{"0xbeefcafe", "3203386110", false},
+		{"0b1010", "10", false},
+		{"12345", "12345", false},
+		{"invalid", "", true},
+		{"0o12345invalid", "", true},
+		{"", "", true},
+	}
+
+	for _, tc := range testCases {
+		result, err := parseIntegerLiteral(tc.input)
```

Comment:
Not a fan of duplicating the code to do a test.

I think we can create a code which runs this test directly into the code of OpEval, by performing `(*gno.Machine).RunStatement`. See `(*gno.Machine).RunMain` for an example of how to call that.

This is the statement we're looking to run: `_ = <literal>`.

Reply:
I'm also not a fan of this method either. However, I couldn't find a proper way to test the `doOpEval` function directly for that moment.

Anyway I think, as you mentioned, create the direct runs by using `(*gno.Machine).RunStatment` is a righteous testing method for this.

P.S. It would be nice to document "how to write test" using this approach on `gnolang/REAME.md`, what do you think?

Reply:
@notJoon let's say the one I mentioned is not actually yet an "official" way to write tests, but rather one that I made up to reasonably make tests here without too much in way of a headache.

As you may have noticed, the gnolang package currently lacks a lot of unit tests, mostly because we fully test it in `tests/files/`, but we should aim to add more and more unit tests which test specific function rather than the entire stack